src/tools/perf/libperf.c

```diff
@@ -767,11 +767,75 @@ static ucs_status_t ucp_perf_test_alloc_iov_mem(ucp_perf_datatype_t datatype,
     return UCS_OK;
 }
 
-static ucs_status_t ucp_perf_test_alloc_mem(ucx_perf_context_t *perf, ucx_perf_params_t *params)
+static ucs_status_t
+ucp_perf_test_alloc_contig(ucx_perf_context_t *perf, ucx_perf_params_t *params,
+                           void **addr, size_t length, ucp_mem_h *memh,
+                           int check_non_blk_flag)
 {
-    ucs_status_t status;
     ucp_mem_map_params_t mem_map_params;
     ucp_mem_attr_t mem_attr;
+    ucs_status_t status;
+
+    if (params->mem_type == UCX_PERF_MEM_TYPE_HOST) {
```

Comment:
i would suggest to break it into 2 new functions (like for gtest): ucp_perf_test_alloc_host and ucp_perf_test_alloc_cuda